rllib/policy/policy.py

```diff
@@ -171,6 +175,79 @@ class Policy(metaclass=ABCMeta):
     `rllib.policy.tf_policy_template::build_tf_policy_class` (TF).
     """
 
+    @staticmethod
+    def from_checkpoint(checkpoint: Union[str, Checkpoint]) -> "Policy":
```

Comment:
I like the approach of having this util function on Policy as a staticmethod.
can we consolidate this with the util we already have? (unfortunately something that I created ...)
https://github.com/ray-project/ray/blob/master/rllib/utils/policy.py#L118

maybe let that util directly call into Policy.from_checkpoint() etc.

Reply:
Ah, didn't know :( 
It's ok, I'll consolidate your util to just use the new `Policy.from_checkpoint` code. We might even consider deprecating it?
There will be extensive documentation on this topic in the next PR, which is already almost done and ready for review (after merging this one). These new docs describe how to use `Algorithm|Policy.from_checkpoint()` in all detail.

Reply:
Actually, now that I'm thinking about this. We should enhance the `Policy.from_checkpoint` by the powers of your util (to extract n Policies from an Algorithm checkpoint!).

For this, I would add the information what type of checkpoint it is: Algorithm or Policy to the checkpoint dir (in the json) and then `Policy.from_checkpoint()` would also work with an Algorithm checkpoint.

Reply:
actually what is the spec now? do policy dirs also get a json file?
I don't know if we need the type field actually. if we provide a dir to ``Algorithm.from_checkpoint()`` or ``Policy.from_checkpoint()``, it's going to be pretty obvious whether the dir contains an Algorithm or Policy checkpoint?

Reply:
I would strongly suggest for policy checkpoints to get a json as well, just for clarity and consistency. I would like to keep `type` in there for now. Again, neither `type` nor `version` are currently used by RLlib, the json file is just a convenience for the user, right now. But we might want to change this in the future.